package/cell_modem_daemon/src/cell_modem_daemon.c

```diff
@@ -175,53 +173,55 @@ int main(int argc, char *argv[])
     /* Prevent czmq from catching signals */
     zsys_handler_set(NULL);
 
-    ctx = sbp_zmq_pubsub_create(SBP_PUB_ENDPOINT, SBP_SUB_ENDPOINT);
-    if (ctx == NULL) {
-      return cleanup(&settings_ctx, &ctx, &port, EXIT_FAILURE);
+    loop = pk_loop_create();
+    if (loop == NULL) {
+      exit(cleanup(&loop, &settings_ctx, &ctx, &port, EXIT_FAILURE));
     }
-    cell_modem_ctx.sbp_ctx = ctx;
 
-    zloop_t *loop = sbp_zmq_pubsub_zloop_get(ctx);
-    if (loop == NULL) {
-      return cleanup(&settings_ctx, &ctx, &port, EXIT_FAILURE);
+    ctx = sbp_pubsub_create(SBP_PUB_ENDPOINT, SBP_SUB_ENDPOINT);
+    if (ctx == NULL) {
+      exit(cleanup(&loop, &settings_ctx, &ctx, &port, EXIT_FAILURE));
     }
+    cell_modem_ctx.sbp_ctx = ctx;
 
-    if (zloop_timer(loop,
-                    CELL_STATUS_UPDATE_INTERVAL,
-                    0,
-                    cell_status_timer_callback,
-                    &cell_modem_ctx)
-        == -1) {
-      return cleanup(&settings_ctx, &ctx, &port, EXIT_FAILURE);
+    if (pk_loop_timer_add(loop,
+                          CELL_STATUS_UPDATE_INTERVAL,
+                          cell_status_timer_callback,
+                          &cell_modem_ctx)
+        == NULL) {
+      exit(cleanup(&loop, &settings_ctx, &ctx, &port, EXIT_FAILURE));
     }
 
     settings_ctx = settings_create();
 
-    cell_modem_init(ctx, settings_ctx);
-
     if (settings_ctx == NULL) {
-      sbp_log(LOG_ERR, "Error registering for settings!");
-      return cleanup(&settings_ctx, &ctx, &port, EXIT_FAILURE);
+      piksi_log(LOG_ERR, "Error registering for settings!");
+      exit(cleanup(&loop, &settings_ctx, &ctx, &port, EXIT_FAILURE));
     }
 
-    if (settings_reader_add(settings_ctx, loop) != 0) {
-      sbp_log(LOG_ERR, "Error registering for settings read!");
-      return cleanup(&settings_ctx, &ctx, &port, EXIT_FAILURE);
+    if (settings_attach(settings_ctx, loop) != 0) {
+      piksi_log(LOG_ERR, "Error registering for settings read!");
+      exit(cleanup(&loop, &settings_ctx, &ctx, &port, EXIT_FAILURE));
     }
 
-    zmq_simple_loop(loop);
+    cell_modem_init(loop, settings_ctx);
+
+    pk_loop_run_simple(loop);
   }
 
-  return cleanup(&settings_ctx, &ctx, &port, EXIT_SUCCESS);
+  exit(cleanup(&loop, &settings_ctx, &ctx, &port, EXIT_SUCCESS));
```

Comment:
Why do we call exit now, instead of return?

Reply:
from what I could tell, both `EXIT_SUCCESS` and `EXIT_FAILURE` are defined for use with `exit()`, I think I am cargo cult-ing the use of `exit` in general. I'm not sure why we use it at all in the general codebase but I figured if we are using its macros we might as well use it